### PR TITLE
Added the add operation for telecom

### PR DESCRIPTION
**Pull request by mani7002**
Description
----
Added Added the add operation for telecom

PR Checklist
----
- [x] Screenshot(s) of changes attached before changes merged.
image
image

- [ ] Screenshot(s) of changes attached after changes merged and published.

### CODE REVIEW

File: lib/resources/r4/related_person_patch.yaml [hunk 1]
```diff
@@ -126,6 +126,45 @@ operations:
...
+                  "period": {
+                      "start": "2018-01-05T22:24:32.000Z"
+                  }
+              }
+          }
+      ]
+    note: >
+      <ul>
+      <li>Must contain the `system`, `use` and `value` fields.</li>
+      <li>Must include a time component with a timezone when specifying a `period`.</li>
```
Comment (Coffeevee) [inline lib/resources/r4/related_person_patch.yaml:165]: Let's be a little more specific with this statement. Another fhir.cerner.com update has this statement about the `period`: https://github.com/cerner/fhir.cerner.com/pull/550/files#diff-a0c01bda612e4656e850590d65d36528f1caec8a70317c2ae4cb71d8095c9e7dR36

Comment (mani7002) [reply lib/resources/r4/related_person_patch.yaml]: Done here:https://github.com/cerner/fhir.cerner.com/pull/559/commits/c8b522d37a1128eb487d1f87a377214d97176251#diff-a0c01bda612e4656e850590d65d36528f1caec8a70317c2ae4cb71d8095c9e7dR165

File: lib/resources/r4/related_person_patch.yaml [hunk 2]
```diff
@@ -126,6 +126,45 @@ operations:
         <li>When specifying a `period`, the `start` and `end` fields must include a time component with a timezone.</li>
       </ul>
 
+  - name: add-telecom
+    path: /telecom/-
+    operation: add
+    type: ContactPoint
+    url: https://hl7.org/fhir/R4/patient-definitions.html#Patient.telecom
```
Comment (SantanuBanerjee) [inline lib/resources/r4/related_person_patch.yaml:133]: should be RelatedPerson.telecom

**Review by Coffeevee — APPROVED** [review]

File: lib/resources/r4/related_person_patch.yaml (hunk 2 above)
Comment (mani7002) [reply lib/resources/r4/related_person_patch.yaml]: Done:https://github.com/cerner/fhir.cerner.com/pull/559/commits/fc371d3cda17ab389ec751373f0a9a0b98dea62d#diff-a0c01bda612e4656e850590d65d36528f1caec8a70317c2ae4cb71d8095c9e7dR133

Comment (SantanuBanerjee) [reply lib/resources/r4/related_person_patch.yaml]: I don't think the link is working yet. Please try and test the link to be working after you change.
I think it should be https://hl7.org/fhir/R4/relatedperson-definitions.html#RelatedPerson. telecom

File: lib/resources/example_json/r4_examples_relatedperson.rb [hunk 3]
```diff
@@ -1371,6 +1371,32 @@ module Resources
           }
         }
       },
+      [
```
Comment (exopsRamba) [inline lib/resources/example_json/r4_examples_relatedperson.rb:1374]: This is already in an array, there shouldn't be another here.

File: lib/resources/r4/related_person_patch.yaml [hunk 4]
```diff
@@ -126,6 +126,45 @@ operations:
...
       </ul>
 
+  - name: add-telecom
+    path: /telecom/-
+    operation: add
+    type: ContactPoint
+    url: https://hl7.org/fhir/R4/related-person-definitions.html#RelatedPerson.telecom
+    description: Add a new contact detail for the related person.
+    example: |
+      [
```
Comment (exopsRamba) [inline lib/resources/r4/related_person_patch.yaml:136]: nit: remove these brackets.

File: lib/resources/r4/related_person_patch.yaml [hunk 5]
```diff
@@ -126,6 +126,45 @@ operations:
...
+                  "period": {
+                      "start": "2018-01-05T22:24:32.000Z"
+                  }
+              }
+          }
+      ]
+    note: >
+      <ul>
+      <li>Must contain the `system`, `use` and `value` fields.</li>
+      <li>When specifying a `period`, the `start` and `end` fields must include a time component with a timezone.</li>
```
Comment (exopsRamba) [inline lib/resources/r4/related_person_patch.yaml:165]: nit: tabbing

Comment (exopsRamba) [inline lib/resources/r4/related_person_patch.yaml:165]: Missing comma after `use`.

File: lib/resources/r4/related_person_patch.yaml [hunk 6]
```diff
@@ -126,6 +126,45 @@ operations:
...
+          	"value": {
+                  "use": "home",
+                  "system": "phone",
+                  "value": "913-444-8881",
+                  "period": {
+                      "start": "2018-01-05T22:24:32.000Z"
+                  }
+              }
+          }
+      ]
```
Comment (exopsRamba) [inline lib/resources/r4/related_person_patch.yaml:161]: nit: tabs should be 2 spaces.

**Review by alexbeckner1 — APPROVED** [review]
+1

File: lib/resources/r4/related_person_patch.yaml (hunk 2 above)
Comment (exopsRamba) [reply lib/resources/r4/related_person_patch.yaml]: Updated here and validated link works: https://github.com/cerner/fhir.cerner.com/pull/559/commits/9a5355ede6a1e2988bb7e041faf5224a1ef4acdb#diff-a0c01bda612e4656e850590d65d36528f1caec8a70317c2ae4cb71d8095c9e7dL133

File: lib/resources/example_json/r4_examples_relatedperson.rb (hunk 3 above)
Comment (sujanb12) [reply lib/resources/example_json/r4_examples_relatedperson.rb]: updated here: https://github.com/cerner/fhir.cerner.com/pull/559/commits/9a5355ede6a1e2988bb7e041faf5224a1ef4acdb#diff-133ac83b73a8a88b430a777cf1700b9010258bd96bc4ff69a51350091f73643eR1374-R1394

File: lib/resources/r4/related_person_patch.yaml (hunk 4 above)
Comment (sujanb12) [reply lib/resources/r4/related_person_patch.yaml]: updated here: https://github.com/cerner/fhir.cerner.com/pull/559/commits/9a5355ede6a1e2988bb7e041faf5224a1ef4acdb#diff-a0c01bda612e4656e850590d65d36528f1caec8a70317c2ae4cb71d8095c9e7dR136-R144

File: lib/resources/r4/related_person_patch.yaml (hunk 5 above)
Comment (sujanb12) [reply lib/resources/r4/related_person_patch.yaml]: updated here: https://github.com/cerner/fhir.cerner.com/pull/559/commits/9a5355ede6a1e2988bb7e041faf5224a1ef4acdb#diff-a0c01bda612e4656e850590d65d36528f1caec8a70317c2ae4cb71d8095c9e7dR162-R163

Comment (sujanb12) [reply lib/resources/r4/related_person_patch.yaml]: added here: https://github.com/cerner/fhir.cerner.com/pull/559/commits/9a5355ede6a1e2988bb7e041faf5224a1ef4acdb#diff-a0c01bda612e4656e850590d65d36528f1caec8a70317c2ae4cb71d8095c9e7dR162

File: lib/resources/r4/related_person_patch.yaml (hunk 6 above)
Comment (sujanb12) [reply lib/resources/r4/related_person_patch.yaml]: updated here: https://github.com/cerner/fhir.cerner.com/pull/559/commits/9a5355ede6a1e2988bb7e041faf5224a1ef4acdb#diff-a0c01bda612e4656e850590d65d36528f1caec8a70317c2ae4cb71d8095c9e7dR136-R159

**Review by ahamdany — APPROVED** [review]

**Review by exopsRamba — APPROVED** [review]

**Review by MaxPhilips — APPROVED** [review]

**Comment by MaxPhilips** [comment]
live
Screen Shot 2021-03-05 at 12 50 42 PM

**Pull request merged**